assets/js/components/ProjectEditorTextField.js

```diff
@@ -0,0 +1,163 @@
+      let url
+      if (this.programSection === 'name') {
+        url = this.pathEditName
+      } else if (this.programSection === 'description') {
+        url = this.pathEditDescription
+      } else if (this.programSection === 'credits') {
+        url = this.pathEditCredits
+      }
+
+      return new Promise(function (resolve, reject) {
+        $.ajax({
+          url: url,
+          type: 'put',
+          data: { value: newText },
+          success: function (data) {
+            const statusCode = parseInt(data.statusCode)
+            if (statusCode === 527 || statusCode === 707) {
+              showError(data)
+              reject(statusCode)
+            } else {
+              resolve(statusCode)
+            }
+          },
+          error: function (error) {
+            if (error.status === 422) {
+              error.message = error.responseText
+              showError(error)
+              reject(error.status)
+            } else if (error.status === 401) {
+              reject(error.status)
+            }
+          }
+        })
+      })
+    } else if (newText === '') {
+      return this.customTranslationApi.deleteCustomTranslation(
+        this.programId,
+        languageSelected,
+        deleteCustomTranslationSuccess,
+        showError
+      )
+    } else {
+      return this.customTranslationApi.saveCustomTranslation(
+        this.programId,
+        newText,
+        languageSelected,
+        saveCustomTranslationSuccess,
+        showError
+      )
+    }
+  }
+
+  function showError (error) {
+    self.editText.addClass('danger')
+    self.editTextError.show()
+    self.editTextError.text(error.message)
+  }
+
+  function deleteCustomTranslationSuccess (language) {
+    self.snackbar.show(self.translationDeleted, language)
+    self.close()
+  }
+
+  function saveCustomTranslationSuccess (language) {
+    self.snackbar.show(self.translationSaved, language)
+    self.close()
+  }
+
+  function getCustomTranslationSuccess (data) {
+    self.textLoadingSpinner.hide()
+    self.editText.val(data)
+    self.lastSavedText = data
+    self.editText.removeAttr('disabled')
+  }
+
+  function getCustomTranslationError () {
+    self.textLoadingSpinner.hide()
+    self.editText.val('')
+    self.lastSavedText = ''
+    self.editText.removeAttr('disabled')
+  }
+
+  this.getNewText = (language) => {
+    if (language === 'default') {
+      this.editText.removeAttr('disabled')
+      this.editText.val(this.initialText)
+      this.lastSavedText = this.initialText
+      return
+    } else if (!hasDefaultText) {
+      this.editText.attr('disabled', '')
+      return
+    }
+
+    this.editText.attr('disabled', '')
+    this.editText.val('')
+    this.textLoadingSpinner.show()
+
+    this.customTranslationApi.getCustomTranslation(
+      this.programId,
+      language,
+      getCustomTranslationSuccess,
+      getCustomTranslationError
+    )
+  }
+
+  this.keepChanges = () => {
```

Comment:
`keepChanges` to me sound like a noop function. Also is it a bug if I do:
change text, then switch language, then select keep changes in the dialog, then press back button, then no "save changes" dialog will be shown?
I'm not sure if this happens, I haven't tried.

Reply:
Updated this keepChanges section to get the saved custom translation or default text as the lastSavedText. Also added some logic to disable or enable text fields